### PR TITLE
Extract the assets from most of the overlays

### PR DESCRIPTION
**Pull request by louist103**
I also updated VTXDIS to use decimal colors and fixed some undefined symbols.

### CODE REVIEW

File: assets/xml/overlays/ovl_Arrow_Fire.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Arrow_Fire.xml:9]: sorry for bringing this up again, I still think this should use "Geometry", not "Vtx",
since the dlist both loads vertices and draws faces (triangles),
and vertex/edge/faces are commonly refered to as "geometry" in 3d modeling

File: assets/xml/overlays/ovl_Arrow_Fire.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Arrow_Fire.xml:8]: in the same vein, this isn't just setting the texture but also setting up the render mode (including blender), combiner and geometry mode,
so I would call this kind of dlist "SetupDL", with the idea it sets up the "drawing parameters" of the geometry that eventually comes next

I would completely agree a texture is more than an image and such a dlist could be viewed as a "texture", but we decided to call images "textures" so for consistency we can't have two definitions of "texture" 😅

File: assets/xml/overlays/ovl_Elf_Msg.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Elf_Msg.xml:11]:
```suggestion
        <Array Name="sElfMsgCubeVtx" Count="8" Offset="0x918">
```

File: assets/xml/overlays/ovl_En_Sth.xml.ignore
Comment (Dragorn421) [inline assets/xml/overlays/ovl_En_Sth.xml.ignore:2]:
```suggestion
ZAPD currently cannot extract regular animations from overlays.    
```

File: assets/xml/overlays/ovl_Magic_Dark.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Magic_Dark.xml:3]: isn't that what `Symbol` is supposed to be for? why does it not work here

File: assets/xml/overlays/ovl_Magic_Wind.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Magic_Wind.xml:13]:
```suggestion
        <DList Name="sMagicWindInnerCylinderDL" Offset="0x19C0"/>
        <DList Name="sMagicWindOuterCylinderDL" Offset="0x1AC8"/>
```

File: src/overlays/actors/ovl_Eff_Dust/z_eff_dust.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Eff_Dust/z_eff_dust.c:39]:
```suggestion
static Gfx sEmptyDL[] = {
```

like in https://github.com/zeldaret/oot/blob/master/src/code/z_rcp.c#L846

File: src/overlays/actors/ovl_En_Ssh/z_en_ssh.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Ssh/z_en_ssh.c:220]: run formatter? surely that line is too long 👀

**Review by Dragorn421 — CHANGES_REQUESTED** [review]
Few cleanup changes are overlapping with #901 and #938 (it's fine but if you keep going we'll end up duplicating efforts needlessly 😅)

File: assets/xml/overlays/ovl_Arrow_Fire.xml
Comment (fig02) [reply assets/xml/overlays/ovl_Arrow_Fire.xml:8]: would it be fair to call that a MaterialDL and the other a ModelDL? I agree with geometry as well but model is a bit shorter and gets the same point across i think

File: format.sh
Comment (Dragorn421) [inline format.sh:12]: you probably didn't mean to commit this

think of plebs like me using clang-format 6 Pepega

but it seems to have formatted some ternaries very nicely 👀 I wish it didnt go away next legacy-clang-format run

Comment (louist103) [reply format.sh:12]: Yeah this was an accident.  but I think we should just pick on a version to use so we dont fight each other with formatting changes.

File: assets/xml/overlays/ovl_Arrow_Fire.xml
Comment (Zelllll) [reply assets/xml/overlays/ovl_Arrow_Fire.xml:9]: sFireArrowShapeDL or sFireArrowTriDL imo

File: assets/xml/overlays/ovl_Arrow_Fire.xml
Comment (Zelllll) [reply assets/xml/overlays/ovl_Arrow_Fire.xml:8]: Material is correct.

**Comment by Dragorn421** [comment]
about Material/Model, those can also change:

`GeometryDL` -> `ModelDL`
`VisualDL` -> `MaterialDL`

File: assets/xml/overlays/ovl_Arrow_Ice.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Arrow_Ice.xml:8]:
```suggestion
        <DList Name="sIceArrowMaterialDL" Offset="0x1CA0"/>
```

File: assets/xml/overlays/ovl_Arrow_Light.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Arrow_Light.xml:8]:
```suggestion
        <DList Name="sLightArrowMaterialDL" Offset="0x1CB0"/>
```

File: assets/xml/overlays/ovl_Magic_Dark.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Magic_Dark.xml:9]:
```suggestion
        <DList Name="sNayrusLoveDiamondModelDL" Offset="0x1700"/>
```
idk, it's used for the other symbols so why not this one too
(may be better: just don't use "Diamond")

File: assets/xml/overlays/ovl_Oceff_Spot.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Oceff_Spot.xml:9]: idk how they should be called but at the very least it should be consistent with similar data (other ovl_Oceff_*), like for song of storms currently:

```xml
        <DList Name="sSongOfStormsCylinderMaterialDL" Offset="0x1A08"/>
        <DList Name="sSongOfStormsCylinderModelDL" Offset="0x1AB0"/>
```
`Material`/`Cylinder` vs `CylinderMaterial`/`CylinderModel`

File: assets/xml/overlays/ovl_Oceff_Wipe4.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Oceff_Wipe4.xml:9]: `Texture` -> `Material`

Comment (Dragorn421) [inline assets/xml/overlays/ovl_Oceff_Wipe4.xml:9]: what about the `@bug` of the "dlist fallthrough" thing that's mentioned in the current _gfx.c
https://github.com/zeldaret/oot/blob/master/src/overlays/actors/ovl_Oceff_Wipe4/z_oceff_wipe4_gfx.c#L100-L126
`sOceff4Texture2DL` also has geometry ("model")

File: src/code/z_parameter.c
Comment (Dragorn421) [inline src/code/z_parameter.c:3782]: idk if this one qualifies as a "material" dlist since it's used on a textured (2d) rectangle

File: src/code/z_rcp.c
Comment (Dragorn421) [inline src/code/z_rcp.c:3]: I thought we would keep Setup instead of Material for this?

File: src/code/z_rcp.c
Comment (Dragorn421) [inline src/code/z_rcp.c:811]: same

File: src/overlays/actors/ovl_En_Holl/z_en_holl.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Holl/z_en_holl.c:328]: ctrl+h gone wrong

File: assets/xml/overlays/ovl_Bg_Ganon_Otyuka.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Bg_Ganon_Otyuka.xml:27]:
```suggestion
        <Collision Name="sCol" Offset="0x2570"/>
```
thonk

File: assets/xml/overlays/ovl_En_Ganon_Organ.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_En_Ganon_Organ.xml:3]: you can probably remove the ganondorf prefix from the OutNames as well

File: assets/xml/overlays/ovl_En_Jsjutan.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_En_Jsjutan.xml:36]:
```suggestion
        <Collision Name="sCol" Offset="0x4C60"/>
```

File: assets/xml/overlays/ovl_Magic_Wind.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Magic_Wind.xml:3]: there's a `d` that you probably didn't mean to put in the symbol names in this xml

File: assets/xml/overlays/ovl_Oceff_spot.xml
Comment (Dragorn421) [inline assets/xml/overlays/ovl_Oceff_spot.xml:5]:
```suggestion
        <Array Name="sCylinderVtx" Count="27" Offset="0xBF0">
```

File: src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c:172]:
```suggestion
        //! @bug If somehow, either of these two actors is not found, the game
        // will dereference a NULL pointer, but this is unlikely to happen.
```
`neither` -> `either`

also don't think this can be considered a bug, as it absolutely cannot be achieved without modding and it's pretty obvious from the code that using this code outside its intended use won't work

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_En_Sth/z_en_sth.c:365]:
```suggestion
        if (this->actor.params != 0) { // Children 
```

File: src/overlays/actors/ovl_Oceff_Wipe4/z_oceff_wipe4.c
Comment (Dragorn421) [inline src/overlays/actors/ovl_Oceff_Wipe4/z_oceff_wipe4.c:106]: why are all of these three dlists named "unused"?

**Review by Dragorn421 — CHANGES_REQUESTED** [review]

File: src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c
Comment (fig02) [reply src/overlays/actors/ovl_En_Jsjutan/z_en_jsjutan.c:172]: agreed, would probably just remove the comment all together personally

File: src/overlays/actors/ovl_En_Sth/z_en_sth.c
Comment (fig02) [reply src/overlays/actors/ovl_En_Sth/z_en_sth.c:365]: this shows the formatter probably needs to be run

**Review by Roman971 — APPROVED** [review]
I'm not sure I agree about changing SetupDL to MaterialDL since material is a lot less explicit imo, but I guess we can sort it out later if needed.

Also the vtxdis changes don't matter to me either since the tool is probably not going to get any use for oot at this point

**Comment by fig02** [comment]
try to avoid adding new content to a pr after it already starts getting reviews and approvals. or at least let us know if you intend to. the previous approvals dont mean anything anymore of more stuff is added that needs review

**Review by fig02 — APPROVED** [review]

**Pull request merged**
